k2/python/host/k2host/fsa.py

```diff
@@ -5,24 +5,26 @@
 import torch
 from torch.utils.dlpack import to_dlpack
 
-from _k2 import IntArray2Size
-from _k2 import _Arc
-from _k2 import DLPackFsa
-from _k2 import IntArray2Size
+from _k2host import IntArray2Size
+from _k2host import _Arc
+from _k2host import DLPackFsa
+from _k2host import IntArray2Size
 
 
 class Arc(_Arc):
 
-    def __init__(self, src_state: int, dest_state: int, label: int):
-        super().__init__(src_state, dest_state, label)
+    def __init__(self, src_state: int, dest_state: int, label: int,
+                 weight: float):
+        super().__init__(src_state, dest_state, label, weight)
 
     def to_tensor(self):
-        return torch.tensor([self.src_state, self.dest_state, self.label],
-                            dtype=torch.int32)
+        return torch.tensor(
+            [self.src_state, self.dest_state, self.label, self.weight],
```

Comment:
isn't the weight accidentally getting truncated from float to int32?

Reply:
Yes, you are right. Another big issue is that `float` has a different binary representation than `int` (except for number 0).

I am looking for a workaround.

Reply:
would it make sense to have `to_tensors` instead and return 2 tensors - one for state ids and labels, and the other one for the weights? I guess that change would affect other APIs too (e.g. `Fsa`).

Reply:
> would it make sense to have to_tensors instead and return 2 tensors

Yes, that's one posibble approach. Another posibble approach is doing type cast, i.e. `float->int` either Python side or C++ side. That was what I did in the python wrapper code before for `LogSumTracebackState::DerivType`  (`std::pair<int, float>`), this approach is also prefered by @danpovey that time. See below code for the details.

https://github.com/k2-fsa/k2/blob/fde770de4b40d20265b5e4d92bbbf6fd53949230/k2/python/csrc/array.cc#L80-L84
https://github.com/k2-fsa/k2/blob/fde770de4b40d20265b5e4d92bbbf6fd53949230/k2/python/csrc/array.cc#L271-L273
https://github.com/k2-fsa/k2/blob/fde770de4b40d20265b5e4d92bbbf6fd53949230/k2/python/k2/array.py#L92-L98
https://github.com/k2-fsa/k2/blob/fde770de4b40d20265b5e4d92bbbf6fd53949230/k2/python/tests/determinize_test.py#L97-L102

note that the last step casts (actually it's `reinterpret`) `float` to `int`, but we usually would not do this frequently, as in model training process, all things related to derivs computation would be done on C++ side, so the reinterpretion is done automatically.